api/test/factory/models/media.py

```diff
@@ -95,6 +96,12 @@ def create(cls, *args, **kwargs) -> AbstractMedia | tuple[AbstractMedia, Hit]:
         skip_es = kwargs.pop("skip_es", False)
         with_hit = kwargs.pop("with_hit", False)
 
+        pook_active = pook.isactive()
+        if pook_active:
+            # Temporarily disable pook so that the calls to ES to create
+            # the factory document don't fail
+            pook.disable()
```

Comment:
This change (and the change later on in this function) makes it so we don't need to juggle pook inside a function to avoid it catching factory requests. Now `@pook.on` can be applied to a whole test function without worry.

I included this in this PR because as part of the ES client's breaking changes, I had to make several updates to tests, mostly removing really flaky and ugly implementation-specific mocks and replacing them with pook matchers. Needing to juggle pook being on or off in all those functions was too tedious.